src/Compiler/Service/TransparentCompiler.fs

```diff
@@ -1575,6 +1601,46 @@ type internal TransparentCompiler
             }
         )
 
+    let TryGetRecentCheckResultsForFile
+        (
+            fileName: string,
+            projectSnapshot: FSharpProjectSnapshot,
+            userOpName: string
+        ) : Async<(FSharpParseFileResults * FSharpCheckFileResults * SourceTextHash) option> =
+        ignore userOpName
+
+        let file =
+            projectSnapshot.ProjectSnapshot.SourceFiles
+            |> List.tryFind (fun f -> f.FileName = fileName)
+
+        match file with
+        | Some f ->
+            async {
+                let! source = f.GetSource() |> Async.AwaitTask
+                let fileKey = projectSnapshot.ProjectSnapshot.FileKey fileName
+                let sourceHash = source.GetHashCode() |> int64
+
+                let cacheKey =
+                    { new ICacheKey<_, _> with
+                        member _.GetLabel() = $"{fileName} ({projectSnapshot.Label})"
+
+                        member _.GetKey() =
+                            fileName, projectSnapshot.ProjectSnapshot.ProjectCore.Identifier
+
+                        member _.GetVersion() = fileKey.GetVersion(), sourceHash
+                    }
+
+                let parseFileResultsAndcheckFileAnswer =
+                    let version = cacheKey.GetVersion()
+                    caches.ParseAndCheckFileInProject.TryGet(cacheKey.GetKey(), (fun v -> v = version))
```

Comment:
I think this should be 

```suggestion
                    caches.ParseAndCheckFileInProject.TryGet(cacheKey.GetKey(), (fun (_, hash) -> hash = sourceHash))
```

                    
Otherwise we're matching the exact version and we wouldn't need the `sourceHash` at all then - and we wouldn't get a result if anything else besides our file was changed.